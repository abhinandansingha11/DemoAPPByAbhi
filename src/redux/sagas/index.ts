import {all, fork} from 'redux-saga/effects';

import todoSaga from './List';

export function* rootSaga() {
  yield all([fork(todoSaga)]);
}
