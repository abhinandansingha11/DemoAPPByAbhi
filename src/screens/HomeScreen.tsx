import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  FlatList,
  Image,
  TouchableOpacity,
} from 'react-native';
import ParentContainer from '../components/ParentContainer';
import {ScreenWidth} from '@freakycoder/react-native-helpers';

import MockData from '../screens/components/card-item/sampleData/SampleData';
import CardItem from '../shared/components/card-item/CardItem';
import AsyncStorage from '@react-native-async-storage/async-storage';
import I18n from '../shared/localization/index';

const profileURI =
  // eslint-disable-next-line max-len
  'https://images.unsplash.com/photo-1544568100-847a948585b9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2574&q=80';

export const HomeScreen = (props: any) => {
  const [user, setUser] = React.useState<any>(MockData);
  React.useEffect(() => {
    _storeData();
  }, []);
  const _storeData = async () => {
    try {
      await AsyncStorage.setItem('name', 'Abhinandan Singha');
      _retrieveData();
    } catch (error) {
      // Error saving data
    }
  };

  const _retrieveData = async () => {
    try {
      const value = await AsyncStorage.getItem('name');
      if (value !== null) {
        // We have data!!
        console.log(value);
      }
    } catch (error) {
      // Error retrieving data
    }
  };

  const List = () => (
    <View style={styles.listContainer}>
      <FlatList data={user} renderItem={({item}) => <CardItem data={item} />} />
    </View>
  );

  const Welcome = () => (
    <>
      <View style={styles.imgStyle}>
        <View>
          <Text>{I18n.t('welcome_Abhi')}</Text>
          <View>
            <Text>{I18n.t('welcome_back')}</Text>
          </View>
        </View>
        <View style={{position: 'absolute', right: 0}}>
          <Image style={styles.tinyLogo} source={{uri: profileURI}} />
        </View>
      </View>
    </>
  );

  const Content = () => (
    <View style={styles.contentContainer}>
      <Welcome />
      <List />
    </View>
  );

  return (
    <ParentContainer {...props}>
      <View style={styles.body}>
        <Content />
      </View>
    </ParentContainer>
  );
};

const styles = StyleSheet.create({
  body: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  container: {
    flex: 1,
    alignItems: 'center',
  },
  titleTextStyle: {
    fontSize: 32,
  },
  buttonStyle: {
    height: 45,
    width: ScreenWidth * 0.9,
    marginTop: 32,
    borderRadius: 12,
    alignItems: 'center',
    justifyContent: 'center',
    shadowRadius: 5,
    shadowOpacity: 0.7,
    shadowOffset: {
      width: 0,
      height: 3,
    },
  },
  buttonTextStyle: {
    fontWeight: '700',
  },
  header: {
    width: ScreenWidth * 0.9,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  contentContainer: {
    flex: 1,
    marginTop: 16,
  },
  listContainer: {
    marginTop: 8,
  },
  profilePicImageStyle: {
    height: 50,
    width: 50,
    borderRadius: 30,
  },
  tinyLogo: {
    width: 50,
    height: 50,
    borderRadius: 25,
    marginRight: 10,
  },
  imgStyle: {
    flexDirection: 'row',
  },
});
