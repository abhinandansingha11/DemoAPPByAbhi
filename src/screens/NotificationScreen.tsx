import I18n from '../shared/localization/index';
import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

import ParentContainer from '../components/ParentContainer';

export const NotificationScreen = (props: any) => {
  return (
    <ParentContainer {...props}>
      <View style={styles.body}>
        <Text>{I18n.t('Notification')}</Text>
      </View>
    </ParentContainer>
  );
};

const styles = StyleSheet.create({
  body: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
