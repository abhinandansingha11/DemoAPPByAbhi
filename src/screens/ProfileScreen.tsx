// MY BAG BOTTOM TAB
import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  ActivityIndicator,
  FlatList,
} from 'react-native';
import {useSelector, useDispatch} from 'react-redux';

import {fetchTodoRequest} from '../redux/actions/List';

import ParentContainer from '../components/ParentContainer';

import {HORIZONTAL_DIMENS} from '../constants/dimens';
import CardItem from './components/card-item/CardItem';
import I18n from '../shared/localization/index';

export const ProfileScreen = (props: any) => {
  const pending = useSelector((state: any) => state.user.pending);
  const todos = useSelector((state: any) => state.user.todos);

  const dispatch = useDispatch();

  React.useEffect(() => {
    dispatch(fetchTodoRequest());
  }, []);

  return (
    <ParentContainer {...props}>
      <View style={styles.body}>
        <Text>{I18n.t('welcome_Abhi')}</Text>
      </View>

      {pending && <ActivityIndicator size={'small'} />}
      <View
        // eslint-disable-next-line react-native/no-inline-styles
        style={{
          justifyContent: 'center',
          alignContent: 'center',
          alignItems: 'center',
          width: '100%',
        }}>
        <FlatList
          data={todos.data}
          renderItem={({item}) => <CardItem data={item} />}
        />
      </View>
    </ParentContainer>
  );
};

const styles = StyleSheet.create({
  body: {
    // flex: 9,
    justifyContent: 'center',
    alignItems: 'center',
    margin: HORIZONTAL_DIMENS._20,
    alignContent: 'center',
    // backgroundColor: 'yellow'
  },

  listItem: {
    width: '100%',
    padding: HORIZONTAL_DIMENS._15,
    backgroundColor: 'grey',
  },
});
