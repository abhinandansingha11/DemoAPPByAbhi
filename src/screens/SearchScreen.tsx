import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {useSelector} from 'react-redux';
import I18n from '../shared/localization/index';

import ParentContainer from '../components/ParentContainer';

export const SearchScreen = (props: any) => {
  const userData = useSelector((state: any) => state.user.user);

  console.log('USER DATA == ', userData);

  return (
    <ParentContainer {...props}>
      <View style={styles.body}>
        <Text>{I18n.t('Search')}</Text>
      </View>
    </ParentContainer>
  );
};

const styles = StyleSheet.create({
  body: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    // backgroundColor: 'yellow'
  },
});
