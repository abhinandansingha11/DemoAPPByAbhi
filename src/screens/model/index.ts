export interface ICardItem {
  first_name: string;
  last_name: string;
  avatar: string;
}
