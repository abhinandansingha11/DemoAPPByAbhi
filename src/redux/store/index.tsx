import {createStore, combineReducers, applyMiddleware} from 'redux';
// import {configureStore} from '@reduxjs/toolkit';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {persistStore, persistReducer} from 'redux-persist';
import createSagaMiddleware from 'redux-saga';
import logger from 'redux-logger';

import ListReducer from '../reducers/List';
import {rootSaga} from '../sagas';

// Create the saga middleware
const sagaMiddleware = createSagaMiddleware();

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

const rootReducer = combineReducers({user: ListReducer});

const configureStore = createStore(
  persistReducer(persistConfig, rootReducer),
  applyMiddleware(sagaMiddleware, logger),
);
sagaMiddleware.run(rootSaga);

export const store = configureStore;
export const persistor = persistStore(store);
