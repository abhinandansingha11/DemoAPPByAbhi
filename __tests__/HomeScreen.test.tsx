import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';

import {store} from '../src/redux/store';
import {HomeScreen} from '../src/screens/HomeScreen';

jest.mock('react-native-localize', () => {
  return {
    getLocales: jest.fn(),
  };
});

jest.mock('@react-native-async-storage/async-storage', () => ({
  setItem: jest.fn(() => Promise.resolve()),
  getItem: jest.fn(() => Promise.resolve()),
}));

jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

it('HomesSreen renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <HomeScreen />
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
