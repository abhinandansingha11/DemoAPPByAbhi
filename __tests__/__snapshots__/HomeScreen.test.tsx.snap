// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomesSreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 16,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View>
          <Text>
            Welcome Abhinandan
          </Text>
          <View>
            <Text>
              Welcome Back
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://images.unsplash.com/photo-1544568100-847a948585b9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2574&q=80",
              }
            }
            style={
              Object {
                "borderRadius": 25,
                "height": 50,
                "marginRight": 10,
                "width": 50,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 8,
          }
        }
      >
        <RCTScrollView
          data={
            Array [
              Object {
                "description": "Demo typescript",
                "language": "Typescript",
                "name": "Abhinandan Singha",
              },
              Object {
                "description": "Demo JS",
                "language": "Typescript",
                "name": "Abhinandan Singha",
              },
              Object {
                "description": "Demo Android",
                "language": "Typescript",
                "name": "Abhinandan Singha",
              },
              Object {
                "description": "Demo Ios",
                "language": "Typescript",
                "name": "Abhinandan Singha",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "borderColor": "#888",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "marginTop": 16,
                    "opacity": 1,
                    "padding": 16,
                    "width": 730,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "https://images.unsplash.com/photo-1544568100-847a948585b9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2574&q=80",
                      }
                    }
                    style={
                      Object {
                        "borderRadius": 25,
                        "height": 50,
                        "marginRight": 10,
                        "width": 50,
                      }
                    }
                  />
                  <Text>
                    Demo typescript
                  </Text>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "borderColor": "#888",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "marginTop": 16,
                    "opacity": 1,
                    "padding": 16,
                    "width": 730,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "https://images.unsplash.com/photo-1544568100-847a948585b9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2574&q=80",
                      }
                    }
                    style={
                      Object {
                        "borderRadius": 25,
                        "height": 50,
                        "marginRight": 10,
                        "width": 50,
                      }
                    }
                  />
                  <Text>
                    Demo JS
                  </Text>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "borderColor": "#888",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "marginTop": 16,
                    "opacity": 1,
                    "padding": 16,
                    "width": 730,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "https://images.unsplash.com/photo-1544568100-847a948585b9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2574&q=80",
                      }
                    }
                    style={
                      Object {
                        "borderRadius": 25,
                        "height": 50,
                        "marginRight": 10,
                        "width": 50,
                      }
                    }
                  />
                  <Text>
                    Demo Android
                  </Text>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "borderColor": "#888",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "marginTop": 16,
                    "opacity": 1,
                    "padding": 16,
                    "width": 730,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "https://images.unsplash.com/photo-1544568100-847a948585b9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2574&q=80",
                      }
                    }
                    style={
                      Object {
                        "borderRadius": 25,
                        "height": 50,
                        "marginRight": 10,
                        "width": 50,
                      }
                    }
                  />
                  <Text>
                    Demo Ios
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
